### ISSUE TITLE
replication/vinyl: data sent on final join are invisible

### ISSUE THREAD

**Issue by locker** [tarantool/tarantool#2002]
Branch 1.7 c689794

Master
```
box.cfg{listen=12345}
box.schema.user.grant('guest', 'read,write,execute', 'universe')
box.schema.space.create('test', {engine = 'vinyl'})
box.space.test:create_index('pk', {parts = {1, 'unsigned'}})
box.space.test:insert{1}
box.space.test:insert{2}
```
Slave
```
tarantool> box.cfg{listen=54321, replication_source=12345, log_level=0}
---
...

tarantool> box.space.test:select()
---
- - [1]
  - [2]
...
```
Restart slave from the snapshot, the records are invisible at first. They reappear after lsn is bumped.
```
tarantool> box.cfg{listen=12123, log_level=0}
---
...

tarantool> box.space.test:select()
---
- []
...
tarantool> for i=1,100 do box.space.test:insert{12345} box.space.test:delete{12345} end
---
...

tarantool> box.space.test:select()
---
- - [1]
  - [2]
...
```

**Comment by rtsisyk** [comment]
Probably fixed  by #2156, please re-check.

**Comment by GeorgyKirichenko** [comment]
Can't reproduce on current 1.7 branch
